index.js

```diff
@@ -141,17 +158,19 @@ MapboxLanguage.prototype._initialStyleUpdate = function () {
   this._map.setStyle(this.setLanguage(style, language));
 };
 
-function browserLanguage(supportedLanguages) {
-  const language = navigator.languages ? navigator.languages[0] : (navigator.language || navigator.userLanguage);
-  const parts = language && language.split('-');
-  let languageCode = language;
-  if (parts.length > 1) {
-    languageCode = parts[0];
-  }
-  if (supportedLanguages.indexOf(languageCode) > -1) {
-    return languageCode;
+function browserLanguage(supportedLanguages, language) {
```

Comment:
This function isn’t recursive anymore, so nothing ever passes in `language` anymore. `language` can go back to being a local constant.